src/test/java/book/store/onlinebookstore/book/BookControllerTest.java

```diff
@@ -0,0 +1,266 @@
+                "Test book",
+                "Test author",
+                "123-123-0001",
+                BigDecimal.valueOf(100),
+                Set.of(1L, 2L),
+                null,
+                null
+        );
+
+        BookDto expected = new BookDto();
+        expected.setId(1L);
+        expected.setTitle(requestDto.title());
+        expected.setAuthor(requestDto.author());
+        expected.setIsbn(requestDto.isbn());
+        expected.setPrice(requestDto.price());
+        expected.setCategoriesIds(requestDto.categoriesIds());
+
+        //when
+        MvcResult result = mockMvc.perform(post("/api/books")
+                        .content(objectMapper.writeValueAsString(requestDto))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        //then
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+        Assertions.assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @DisplayName("Check if get all books returns list of books dto")
+    @WithMockUser
+    @Sql(scripts = "classpath:database.scripts/book/add-three-books.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database.scripts/book/clear-book-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @SneakyThrows
+    void getAll_ValidPageable_ReturnsListOfBooks() {
+        //given
+        BookDto book1 = new BookDto();
+        book1.setId(1L);
+        book1.setTitle("test-book1");
+        book1.setAuthor("test-author");
+        book1.setIsbn("123-123-0001");
+        book1.setPrice(BigDecimal.valueOf(100.99));
+        book1.setCategoriesIds(Set.of());
+        BookDto book2 = new BookDto();
+        book2.setId(2L);
+        book2.setTitle("test-book2");
+        book2.setAuthor("test-author");
+        book2.setIsbn("123-123-0002");
+        book2.setPrice(BigDecimal.valueOf(200.99));
+        book2.setCategoriesIds(Set.of());
+        BookDto book3 = new BookDto();
+        book3.setId(3L);
+        book3.setTitle("test-book3");
+        book3.setAuthor("test-author");
+        book3.setIsbn("123-123-0003");
+        book3.setPrice(BigDecimal.valueOf(300.99));
+        book3.setCategoriesIds(Set.of());
+        List<BookDto> expected = List.of(book1, book2, book3);
+
+        //when
+        MvcResult result = mockMvc.perform(get("/api/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //then
+        BookDto[] actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto[].class);
+        Assertions.assertEquals(expected.size(), actual.length);

```

Comment:
it's good practice to use static import 
```suggestion
      assertEquals(expected.size(), actual.length);
```